packages/swingset-liveslots/test/virtual-objects/test-cross-facet.js

```diff
@@ -0,0 +1,58 @@
+import test from 'ava';
+import '@endo/init/debug.js';
+
+import { makeFakeVirtualObjectManager } from '../../tools/fakeVirtualSupport.js';
+
+// This tests the line in `makeContextProviderKit()` which guards
+// against the prototype from one facet being called with a 'this' that
+// refers to a different facet:
+//    Number(facet) === index || Fail`illegal cross-facet access`;
+//
+// Without this check, an attacker (with access to both cohort1.facetA
+// and cohort2.facetB) could effectively forge access to cohort1.facetB
+// and cohort2.facetA. They could not forge the identity of those two
+// objects, but they could invoke all their equivalent methods, by using
+// e.g. cohort1.facetA.foo.apply(cohort2.facetB, [...args])
+//
+// Exercise this by commenting out that check and confirming that the
+// test now fails.
+
+function attack1(mut1, immut2) {
+  mut1.set.apply(immut2, [5]);
+}
+
+function attack2(mut1, immut2) {
+  const mutableProto = Object.getPrototypeOf(mut1);
+  Reflect.apply(mutableProto.set, immut2, [6]);
+}
+
+test('forbid cross-facet prototype attack', async t => {
+  const vom = makeFakeVirtualObjectManager();
+  const init = () => ({ value: 0 });
+  const behavior = {
+    mutable: {
+      set: ({ state }, value) => (state.value = value),
+      get: ({ state }) => state.value,
+    },
+    immutable: {
+      get: ({ state }) => state.value,
+    },
+  };
+  const makeThing = vom.defineKindMulti('thing', init, behavior);
+  const thing1 = makeThing();
+  thing1.mutable.set(1);
+  const thing2 = makeThing();
+  thing2.mutable.set(2);
+
+  const m = s => ({ message: s });
+  t.throws(
+    () => attack1(thing1.mutable, thing2.immutable),
+    m(/^illegal cross-facet access/),
+  );
+  t.throws(
+    () => attack2(thing1.mutable, thing2.immutable),
+    m(/^illegal cross-facet access/),
+  );
+  t.is(thing1.immutable.get(), 1);
+  t.is(thing2.immutable.get(), 2);
```

Comment:
Is this `m` really useful?
```suggestion
  t.throws(
    () => attack1(thing1.mutable, thing2.immutable),
    { message: /^illegal cross-facet access/ },
  );
  t.throws(
    () => attack2(thing1.mutable, thing2.immutable),
    { message: /^illegal cross-facet access/ },
  );
  t.is(thing1.immutable.get(), 1);
  t.is(thing2.immutable.get(), 2);
```

Reply:
good point, I think I was using larger error messages before, instead of just a regexp for the most important bit, making the lines wider

Reply:
included in https://github.com/Agoric/agoric-sdk/pull/7307